service/src/main/java/org/dice_research/fc/sparql/query/QueryEngineCustomHTTP.java

```diff
@@ -0,0 +1,225 @@
+package org.dice_research.fc.sparql.query;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.ByteSource;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.query.*;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.sparql.core.Quad;
+import org.apache.jena.sparql.util.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class run SPARQL queries with CloseableHttpClient .
+ *
+ * @author Farshad Afshari
+ *
+ */
+
+public class QueryEngineCustomHTTP implements QueryExecution {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryEngineCustomHTTP.class);
+
+    private Query query;
+    private String service;
+    private long timeout=15000;
```

Comment:
Why is the timeout set to 15 seconds (or something similar) by default? :open_mouth: 

1. Such a behavior has to be documented in the class description. Otherwise, it will lead to a lot of confusion for people that may have issues with this timeout.
2. As a user of such a class, I would expect _no timeout_ by default. :thinking: